src/System.Collections/tests/Performance/Perf.Dictionary.cs

```diff
@@ -0,0 +1,195 @@
+                dict.Add("key" + i, "value");
+
+            // Actual perf testing
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    retrieved = dict["key1"]; retrieved = dict["key2"]; retrieved = dict["key3"];
+                    retrieved = dict["key4"]; retrieved = dict["key5"]; retrieved = dict["key6"];
+                    retrieved = dict["key7"]; retrieved = dict["key8"]; retrieved = dict["key9"];
+                }
+            }
+
+            // Teardown
+            for (int i = 1; i <= 9; i++)
+                dict.Remove("key" + i);
+        }
+
+        [Benchmark]
+        [MemberData("TestDictionaries")]
+        public void SetItem(Dictionary<string, string> dict)
+        {
+            // Setup
+            for (int i = 1; i <= 9; i++)
+                dict.Add("key" + i, "value");
+
+            // Actual perf testing
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    dict["key1"] = "string"; dict["key2"] = "string"; dict["key3"] = "string";
+                    dict["key4"] = "string"; dict["key5"] = "string"; dict["key6"] = "string";
+                    dict["key7"] = "string"; dict["key8"] = "string"; dict["key9"] = "string";
+                }
+            }
+
+            // Teardown
+            for (int i = 1; i <= 9; i++)
+                dict.Remove("key" + i);
+        }
+
+        [Benchmark]
+        [MemberData("TestDictionaries")]
+        public void GetKeys(Dictionary<string, string> dict)
+        {
+            IEnumerable<string> result;
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                {
+                    result = dict.Keys; result = dict.Keys; result = dict.Keys;
+                    result = dict.Keys; result = dict.Keys; result = dict.Keys;
+                    result = dict.Keys; result = dict.Keys; result = dict.Keys;
+                }
+        }
+
+        [Benchmark]
+        [MemberData("TestDictionaries")]
+        public void TryGetValue(Dictionary<string, string> dict)
+        {
+            // Setup
+            string retrieved;
+            dict.Add("key", "value");
+
+            // Actual perf testing
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    dict.TryGetValue("key", out retrieved); dict.TryGetValue("key", out retrieved);
+                    dict.TryGetValue("key", out retrieved); dict.TryGetValue("key", out retrieved);
+                    dict.TryGetValue("key", out retrieved); dict.TryGetValue("key", out retrieved);
+                    dict.TryGetValue("key", out retrieved); dict.TryGetValue("key", out retrieved);
```

Comment:
I like that idea. I could add a custom string class that uses the same GetHashCode/Equals but adds a fixed delay to them so that we have a good count of the number of times the functions are called.

A simpler approach would be to change the keys the be randomly generated using PerfUtils.CreateString(50) so that they take longer to Equals/GetHashCode, but that may not take long enough.